dashboard/app/controllers/maker_controller.rb

```diff
@@ -39,6 +43,12 @@ def self.any_csd_2018?(for_user)
       for_user.section_courses.include?(Course.get_from_cache(ScriptConstants::CSD_2018))
   end
 
+  def self.any_csd_2019?(for_user)
+    !for_user.user_scripts.joins(:script).
+      where(scripts: {name: ScriptConstants::CATEGORIES[:csd_2019]}).empty? ||
+      for_user.section_courses.include?(Course.get_from_cache(ScriptConstants::CSD_2019))
+  end
```

Comment:
I look briefly at deduping this against the above methods, but we don't have an existing mapping of year -> CSD course. I can add one if necessary.